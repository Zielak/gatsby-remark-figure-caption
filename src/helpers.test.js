const assert = require("mdast-util-assert");
const render = require("unist-util-inspect");
const { createElement, replace } = require("./helpers");

/** @jsx createElement */

expect.addSnapshotSerializer({
	test: (tree) => tree && Array.isArray(tree.children) && tree.type,
	print: (tree) => render.noColor(tree),
});

const KitchenSink = ({ name = "", ...props }) => (
	<blockquote {...props}>
		<paragraph>wat</paragraph>
		<image className={""} />
		{false && "wont render"}
		{0}
		{undefined}
		{null}
		<image url="http://" />
		{name.split("").map((letter) => (
			<paragraph>{letter}</paragraph>
		))}
	</blockquote>
);

describe("createElement", () => {
	it("renders custom component", () => {
		const mdast = <KitchenSink tagName="section" name="batman" />;
		expect(mdast).toMatchSnapshot();
		expect(() => assert(mdast)).not.toThrow();
	});
});

describe("replace(source).with(target)", () => {
	const source = { alt: "lazy stuff" },
		sourceBackup = { ...source };
	const target = { tagName: "figure" },
		targetBackup = { ...target };
	const result = replace(source).with(target);

	it("remove source props that arent in the target", () => {
		expect(source).toEqual({ tagName: "figure" });
	});

	it("should not mutate the target", () => {
		expect(target).toEqual(targetBackup);
	});

	it("should mutate the source", () => {
		expect(source).not.toEqual(sourceBackup);
	});

	it("should return undefined", () => {
		expect(result).toBeUndefined();
	});
});
