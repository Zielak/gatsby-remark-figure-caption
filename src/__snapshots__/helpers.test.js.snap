// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createElement renders custom component 1`] = `
blockquote[9]
├─ paragraph[1]
│  └─ text: "wat"
├─ image
├─ image [url="http://"]
├─ paragraph[1]
│  └─ text: "b"
├─ paragraph[1]
│  └─ text: "a"
├─ paragraph[1]
│  └─ text: "t"
├─ paragraph[1]
│  └─ text: "m"
├─ paragraph[1]
│  └─ text: "a"
└─ paragraph[1]
   └─ text: "n"
`;
